IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -114,14 +119,31 @@ + (nullable MSIDWPJKeyPairWithCert *)findWPJRegistrationInfoWithAdditionalPrivat
     queryPrivateKey[(__bridge id)kSecClass] = (__bridge id)kSecClassKey;
     queryPrivateKey[(__bridge id)kSecReturnAttributes] = @YES;
     queryPrivateKey[(__bridge id)kSecReturnRef] = @YES;
+#if !MSID_ENABLE_ECC_SUPPORT
     // TODO: hardcoding this to query RSA keys only for now. Once ECC registration is ready and tested, after removing this line, code should be able to find either ECC or RSA keys, since there should be single key corresponding to the tag per tenant
     queryPrivateKey[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    
+#endif
     status = SecItemCopyMatching((__bridge CFDictionaryRef)queryPrivateKey, (CFTypeRef*)&privateKeyCFDict); // +1 privateKeyCFDict
     if (status != errSecSuccess)
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to find workplace join private key with status %ld", (long)status);
-        return nil;
+        if (shouldCheckEnclave && privateKeyCFDict == NULL)
+        {
+            // Checking if key exists in Secure Enclave
+            NSData *tagData = queryPrivateKey[(__bridge id)kSecAttrApplicationTag];
```

Comment:
I'm not sure the logic in here is correct, maybe I misunderstood:
get tagdata from privatekey dict query
get tag string from tag data
if tag string doesn't end with ecc suffix:  create new tag string, then add back to private key dict query.

Why the check needed, should you just add the correct tag to dict query anyway?

Reply:
Yes, updated the logic to re-call this method with correct tag instead of doing it here.